### ISSUE TITLE
Allow multiple deferred.resolve() parameters

### ISSUE THREAD

**Issue by enyo** [kriskowal/q#92]
Maybe I'm not seeing something but wouldn't it be cool to support multiple `deferred.resolve()` parameters?

You could then call `deferred.resolve(a, b, c);` in your API and in your code: `doStuff().then(a, b, c);`

**Comment by mmocny** [comment]
Sorry to resurrect an old bug.. but I just spent a bit of time tracking down why `nfcall` [returns an array of results](https://github.com/kriskowal/q/blob/v1/q.js#L621) instead of applying them as arguments to callback.

For the record, I agree with the decision now that I know the philosophy and can see why this decision helps with many use cases (e.g. `spawn()` type generator based workflows).

However, I did not catch this reasoning from reading your `README.md` (many times).  Maybe I just missed it, but the [specs for `nfcall`](https://github.com/kriskowal/q#adapting-node) etc do not make it obvious.
